enhancements/storage/gcp-filestore-wif.md

```diff
@@ -0,0 +1,231 @@
+
+      | Dashboard name        | CR field name                         | Env var name          |
+      |-----------------------|---------------------------------------|-----------------------|
+      | GCP Tenant ID         | spec.providerSpec.poolID              | POOL_ID               |
+      | GCP Provider ID       | spec.providerSpec.providerID          | PROVIDER_ID           |
+      | Service Account Email | spec.providerSpec.serviceAccountEmail | SERVICE_ACCOUNT_EMAIL |
+   
+    - The value for `Audience` has to be created based on `PROJECT_NUMBER`, `POOL_ID`, `PROVIDER_ID` environment variables, and has to validate against regex check defined by CCO: [openshift/cloud-credential-operator#708](https://github.com/openshift/cloud-credential-operator/pull/708).
+
+
+3. **RBAC Permissions**:
+    - Ensure the operator has the necessary RBAC permissions to create `CredentialsRequest` (Document: [Google Doc](https://docs.google.com/document/d/1iFNpyycby_rOY1wUew-yl3uPWlE00krTgr9XHDZOTNo/edit#heading=h.vzrrddcdj5gj)).
+    - This is a CSV update only.
+
+
+4. **Volume and Mount Definition**:
+    - Define the required volume and mount for bound service account token in the CSV (Document: [Google Doc](https://docs.google.com/document/d/1iFNpyycby_rOY1wUew-yl3uPWlE00krTgr9XHDZOTNo/edit#heading=h.vzrrddcdj5gj)).
+    - This is a CSV update only.
+
+
+5. **WIF Support Announcement**:
+    - Update the CSV to include the annotation: `metadata.annotations.features.operators.openshift.io/token-auth-gcp: "true"` (Document: [Google Doc](https://docs.google.com/document/d/1iFNpyycby_rOY1wUew-yl3uPWlE00krTgr9XHDZOTNo/edit#heading=h.vzrrddcdj5gj)).
+    - This is a CSV update only.
+
+
+6. **Handling Secrets**:
+    - Ensure the GCP Filestore operator waits for the secret to be created by CCO and provide meaningful error messages if the secret is not available.
+
+
+### Workflow Description
+
+1. User locates GCP Filestore in the OperatorHub and clicks the operator tile.
+2. User clicks "Install" and is presented with installation options.
+3. The form presented to user includes fields for the new `CredentialsRequest` values - `ServiceAccountEmail`, `PoolID` and `ProviderID`.
+4. User fills in the required fields and clicks "Install".
+5. OLM places the values into the `Subscription` spec configuration which are injected into GCP Filestore Operator pods as environment variables - `PROJECT_NUMBER`, `POOL_ID`, `PROVIDER_ID`, `SERVICE_ACCOUNT_EMAIL`
+6. The operator creates `CredentialsRequest` resource using the provided values.
+7. CCO processes the `CredentialsRequest` and creates a secret with the necessary credentials.
+8. GCP Filestore driver uses the secret to authenticate with GCP services with short-lived credentials.
+
+### API Extensions
+
+- No modifications to existing storage resources.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+- Ensure WIF functionality is compatible with Hypershift deployments.
```

Comment:
Do we support hypershift with gcp file store or gcp-pd deployments?

Reply:
GCP is not yet supported with hosted control plane - removing.